### PR TITLE
add free scale graphs generation

### PR DESCRIPTION
**Pull request by simpletonDL**

### CODE REVIEW

File: init.py [hunk 1]
```diff
@@ -127,7 +128,32 @@ def gen_cycle_graph(target_dir, vertices):
...
         for i in range(0, vertices - 1):
             out_file.write('%s A %s \n'%(i, i + 1))
-        out_file.write('%s A %s \n'%(vertices - 1, 0))   
+        out_file.write('%s A %s \n'%(vertices - 1, 0))
+
+def gen_free_scale_graph(target_dir, n, k, labels, reverse_edges=False):
+    g = {
+        i: [(j, random.choice(labels))
+            for j in range(k)] for i in range(k)
+    }
```
Comment (ciromoraismedeiros) [inline init.py]: Suggestion: use a 3xn matrix to store the graph. In this way you can use the _to_file function. Moreover, the triples will be written to the file in the insertion order, whereas with the dictionary they won't.
I believe it's not gonna change much in the overall logic.

File: data/graphs/FreeScale/Grammars/an_bm_cm_dn.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+S1 S D
+S A S2
+S2 X D
+X B C
+X B X1
+X1 X C
+A a
+B b
+C c
+D d
```
Comment (ciromoraismedeiros) [inline data/graphs/FreeScale/Grammars/an_bm_cm_dn.txt]: It necessary to add also the original non-CNF version of grammars for non-CNF-based algorithms.
CNF increase grammar's size, what affects algorithms' performance.

File: init.py (hunk 1 above)
Comment (simpletonDL) [reply init.py]: Ok, I will change it. But I cant open source code of `_to_file` function (may be due to force-push). Now I want to merge this pull request and make refactoring which you suggested.

File: data/graphs/FreeScale/Grammars/an_bm_cm_dn.txt (hunk 2 above)
Comment (simpletonDL) [reply data/graphs/FreeScale/Grammars/an_bm_cm_dn.txt]: I will add this to the refactoring commit.

**Comment by simpletonDL** [comment]
Thank you for your review!) I will fix it soon.

**Pull request merged**